#Importação

In [ ]:
import pandas as pd

In [ ]:
dados = pd.read_csv('datatran2020.csv', delimiter=';')
dados.head()

id data_inversa dia_semana   horario  uf     br     km    municipio  \
0  260031.0   2020-01-01     quarta  01:00:00  TO  153.0  678,1       GURUPI   
1  260036.0   2020-01-01     quarta  01:00:00  RJ  116.0    178  NOVA IGUACU   
2  260037.0   2020-01-01     quarta  01:52:00  SC  101.0  206,9     SAO JOSE   
3  260038.0   2020-01-01     quarta  01:15:00  RO  364.0    236       CACOAL   
4  260039.0   2020-01-01     quarta  01:30:00  PR  153.0    360     REBOUCAS   

                 causa_acidente              tipo_acidente  ...  \
0              Animais na Pista    Atropelamento de Animal  ...   
1  Falta de Atenção do Pedestre  Atropelamento de Pedestre  ...   
2            Ingestão de Álcool            Colisão lateral  ...   
3       Velocidade Incompatível  Saída de leito carroçável  ...   
4            Ingestão de Álcool                Capotamento  ...   

  feridos_graves ilesos ignorados feridos veiculos      latitude  \
0              5      0         0       5        1  -11,77460203   
1              0      1         0       0        1  -22,75223028   
2              0      1         0       1        2  -27,59193546   
3              1      0         0       1        1  -11,44624577   
4              4      0         0       6        1  -25,67503796   

      longitude  regional  delegacia       uop  
0  -49,10744996     SR-TO   UOP01/TO       NaN  
1   -43,4379103     SR-RJ     DEL5/1  UOP02/RJ  
2  -48,61824557     SR-SC     DEL8/1  UOP01/SC  
3  -61,43761218     SR-RO    DEL21/2  UOP02/RO  
4  -50,75089805     SR-PR     DEL7/3  UOP04/PR  

[5 rows x 30 columns]

In [ ]:
dados.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 63530 entries, 0 to 63529
Data columns (total 30 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   id                      63530 non-null  float64
 1   data_inversa            63530 non-null  object 
 2   dia_semana              63530 non-null  object 
 3   horario                 63530 non-null  object 
 4   uf                      63530 non-null  object 
 5   br                      63372 non-null  float64
 6   km                      63372 non-null  object 
 7   municipio               63530 non-null  object 
 8   causa_acidente          63530 non-null  object 
 9   tipo_acidente           63530 non-null  object 
 10  classificacao_acidente  63530 non-null  object 
 11  fase_dia                63530 non-null  object 
 12  sentido_via             63530 non-null  object 
 13  condicao_metereologica  63530 non-null  object 
 14  tipo_pista              63530 non-null

# Pré-processamento

## Tratando as datas com datetime

Queremos apenas saber o mês, dado que a planilha é sobre o ano, e o dia do mês pouco importa, dado que o dia da semana está explicitado em outra coluna

Necessário usar:
- pd.to_datetime
- dt.strftime("%d-%m-%Y")

In [ ]:
# 1. Converte tudo para datetime
dados.data_inversa = pd.to_datetime(dados.data_inversa)

# 2. Muda o formato de toda a coluna
dados.data_inversa = dados.data_inversa.dt.strftime("%m")

dados.data_inversa.info()

<class 'pandas.core.series.Series'>
RangeIndex: 63530 entries, 0 to 63529
Series name: data_inversa
Non-Null Count  Dtype 
--------------  ----- 
63530 non-null  object
dtypes: object(1)
memory usage: 496.5+ KB


In [ ]:
datas = {
    '01' : 'Janeiro',
    '02' : 'Fevereiro',
    '03' : 'Março',
    '04' : 'Abril',
    '05' : 'Maio',
    '06' : 'Junho',
    '07' : 'Julho',
    '08' : 'Agosto',
    '09' : 'Setembro',
    '10' : 'Outubro',
    '11' : 'Novembro',
    '12' : 'Dezembro'
}

In [ ]:
dados.data_inversa = dados.data_inversa.map(datas)
dados.data_inversa.head()

0    Janeiro
1    Janeiro
2    Janeiro
3    Janeiro
4    Janeiro
Name: data_inversa, dtype: object

Por fim, devemos fazer a troca do nome da coluna usando a função **rename**

In [ ]:
dados = dados.rename(columns={'data_inversa' : 'mes'})
dados.mes.head()

0    Janeiro
1    Janeiro
2    Janeiro
3    Janeiro
4    Janeiro
Name: mes, dtype: object

## Limpando colunas inúteis

In [ ]:
dados.head()

id      mes dia_semana   horario  uf     br     km    municipio  \
0  260031.0  Janeiro     quarta  01:00:00  TO  153.0  678,1       GURUPI   
1  260036.0  Janeiro     quarta  01:00:00  RJ  116.0    178  NOVA IGUACU   
2  260037.0  Janeiro     quarta  01:52:00  SC  101.0  206,9     SAO JOSE   
3  260038.0  Janeiro     quarta  01:15:00  RO  364.0    236       CACOAL   
4  260039.0  Janeiro     quarta  01:30:00  PR  153.0    360     REBOUCAS   

                 causa_acidente              tipo_acidente  ...  \
0              Animais na Pista    Atropelamento de Animal  ...   
1  Falta de Atenção do Pedestre  Atropelamento de Pedestre  ...   
2            Ingestão de Álcool            Colisão lateral  ...   
3       Velocidade Incompatível  Saída de leito carroçável  ...   
4            Ingestão de Álcool                Capotamento  ...   

  feridos_graves ilesos ignorados feridos veiculos      latitude  \
0              5      0         0       5        1  -11,77460203   
1              0      1         0       0        1  -22,75223028   
2              0      1         0       1        2  -27,59193546   
3              1      0         0       1        1  -11,44624577   
4              4      0         0       6        1  -25,67503796   

      longitude  regional  delegacia       uop  
0  -49,10744996     SR-TO   UOP01/TO       NaN  
1   -43,4379103     SR-RJ     DEL5/1  UOP02/RJ  
2  -48,61824557     SR-SC     DEL8/1  UOP01/SC  
3  -61,43761218     SR-RO    DEL21/2  UOP02/RO  
4  -50,75089805     SR-PR     DEL7/3  UOP04/PR  

[5 rows x 30 columns]

In [ ]:
dados.columns

Index(['id', 'mes', 'dia_semana', 'horario', 'uf', 'br', 'km', 'municipio',
       'causa_acidente', 'tipo_acidente', 'classificacao_acidente', 'fase_dia',
       'sentido_via', 'condicao_metereologica', 'tipo_pista', 'tracado_via',
       'uso_solo', 'pessoas', 'mortos', 'feridos_leves', 'feridos_graves',
       'ilesos', 'ignorados', 'feridos', 'veiculos', 'latitude', 'longitude',
       'regional', 'delegacia', 'uop'],
      dtype='object')

Queremos retirar todas as colunas irrelevantes para a análise desejada. Com isso, podemos retirar as que possuem dados relacionados ao local específico, delegacia de ocorrência, dentre outros.

In [ ]:
dados.drop(columns=['id', 'br', 'km',
                    'latitude', 'longitude', 'regional',
                    'delegacia', 'uop', 'sentido_via', 'municipio'], inplace=True)
dados.head()

mes dia_semana   horario  uf                causa_acidente  \
0  Janeiro     quarta  01:00:00  TO              Animais na Pista   
1  Janeiro     quarta  01:00:00  RJ  Falta de Atenção do Pedestre   
2  Janeiro     quarta  01:52:00  SC            Ingestão de Álcool   
3  Janeiro     quarta  01:15:00  RO       Velocidade Incompatível   
4  Janeiro     quarta  01:30:00  PR            Ingestão de Álcool   

               tipo_acidente classificacao_acidente     fase_dia  \
0    Atropelamento de Animal    Com Vítimas Feridas  Plena Noite   
1  Atropelamento de Pedestre     Com Vítimas Fatais  Plena Noite   
2            Colisão lateral    Com Vítimas Feridas  Plena Noite   
3  Saída de leito carroçável    Com Vítimas Feridas  Plena Noite   
4                Capotamento    Com Vítimas Feridas  Plena Noite   

  condicao_metereologica tipo_pista tracado_via uso_solo  pessoas  mortos  \
0                Nublado    Simples        Reta      Não        5       0   
1              Céu Claro      Dupla        Reta      Sim        2       1   
2                Nublado    Simples        Reta      Sim        2       0   
3                Nublado    Simples        Reta      Sim        1       0   
4         Garoa/Chuvisco    Simples       Curva      Não        6       0   

   feridos_leves  feridos_graves  ilesos  ignorados  feridos  veiculos  
0              0               5       0          0        5         1  
1              0               0       1          0        0         1  
2              1               0       1          0        1         2  
3              0               1       0          0        1         1  
4              2               4       0          0        6         1

# Análise e exploração dos dados

In [ ]:
dados.head()

mes dia_semana   horario  uf                causa_acidente  \
0  Janeiro     quarta  01:00:00  TO              Animais na Pista   
1  Janeiro     quarta  01:00:00  RJ  Falta de Atenção do Pedestre   
2  Janeiro     quarta  01:52:00  SC            Ingestão de Álcool   
3  Janeiro     quarta  01:15:00  RO       Velocidade Incompatível   
4  Janeiro     quarta  01:30:00  PR            Ingestão de Álcool   

               tipo_acidente classificacao_acidente     fase_dia  \
0    Atropelamento de Animal    Com Vítimas Feridas  Plena Noite   
1  Atropelamento de Pedestre     Com Vítimas Fatais  Plena Noite   
2            Colisão lateral    Com Vítimas Feridas  Plena Noite   
3  Saída de leito carroçável    Com Vítimas Feridas  Plena Noite   
4                Capotamento    Com Vítimas Feridas  Plena Noite   

  condicao_metereologica tipo_pista tracado_via uso_solo  pessoas  mortos  \
0                Nublado    Simples        Reta      Não        5       0   
1              Céu Claro      Dupla        Reta      Sim        2       1   
2                Nublado    Simples        Reta      Sim        2       0   
3                Nublado    Simples        Reta      Sim        1       0   
4         Garoa/Chuvisco    Simples       Curva      Não        6       0   

   feridos_leves  feridos_graves  ilesos  ignorados  feridos  veiculos  
0              0               5       0          0        5         1  
1              0               0       1          0        0         1  
2              1               0       1          0        1         2  
3              0               1       0          0        1         1  
4              2               4       0          0        6         1

## Análise simples

### Qual a causa de acidentes mais frequente?

In [ ]:
dados.causa_acidente.unique()

array(['Animais na Pista', 'Falta de Atenção do Pedestre',
       'Ingestão de Álcool', 'Velocidade Incompatível',
       'Falta de Atenção à Condução',
       'Desobediência às normas de trânsito pelo condutor',
       'Condutor Dormindo', 'Pista Escorregadia', 'Mal Súbito',
       'Avarias e/ou desgaste excessivo no pneu',
       'Não guardar distância de segurança',
       'Deficiência ou não Acionamento do Sistema de Iluminação/Sinalização do Veículo',
       'Ingestão de álcool e/ou substâncias psicoativas pelo pedestre',
       'Defeito Mecânico no Veículo', 'Ultrapassagem Indevida',
       'Defeito na Via', 'Ingestão de Substâncias Psicoativas',
       'Sinalização da via insuficiente ou inadequada',
       'Objeto estático sobre o leito carroçável', 'Agressão Externa',
       'Fenômenos da Natureza', 'Restrição de Visibilidade',
       'Desobediência às normas de trânsito pelo pedestre',
       'Carga excessiva e/ou mal acondicionada',
       'Ausência de reação do condutor', '

In [ ]:
mais_frequente = dados.causa_acidente.value_counts()
mais_frequente.head(5)

Falta de Atenção à Condução                          22410
Desobediência às normas de trânsito pelo condutor     8802
Velocidade Incompatível                               5883
Ingestão de Álcool                                    5069
Defeito Mecânico no Veículo                           3742
Name: causa_acidente, dtype: int64

Portanto, os maiores motivo dos acidentes de trânsito no ano de 2020 foram:

- Falta de atenção
- Desobediência das normas de trânsito
- Velocidade incompativel
- Ingestão de álcool
- Defeito mecânico no veículo

In [ ]:
dados.causa_acidente.info()

<class 'pandas.core.series.Series'>
RangeIndex: 63530 entries, 0 to 63529
Series name: causa_acidente
Non-Null Count  Dtype 
--------------  ----- 
63530 non-null  object
dtypes: object(1)
memory usage: 496.5+ KB


Assim, temos 63530 dados não nulos nessa coluna. Vamos pegar as frequências relativas dos primeiros 5 maiores motivos de acidentes.


In [ ]:
contagem = dados.causa_acidente.count()
f1 = (22410 / contagem) * 100
f2 = (8802 / contagem) * 100
f3 = (5883 / contagem) * 100
f4 = (5069 / contagem) * 100
f5 = (3742 / contagem) * 100

print(f'Frequência relativa - Falta de atenção: {f1:.2f}%')
print(f'Frequência relativa - Desobediência das normas de trânsito: {f2:.2f}%')
print(f'Frequência relativa - Velocidade incompatível: {f3:.2f}%')
print(f'Frequência relativa - Ingestão de álcool: {f4:.2f}%')
print(f'Frequência relativa - Defeitos mecânicos no veículo: {f5:.2f}%')

Frequência relativa - Falta de atenção: 35.27%
Frequência relativa - Desobediência das normas de trânsito: 13.85%
Frequência relativa - Velocidade incompatível: 9.26%
Frequência relativa - Ingestão de álcool: 7.98%
Frequência relativa - Defeitos mecânicos no veículo: 5.89%


In [ ]:
soma_maiores = f1 + f2 + f3 + f4 + f5
print(f'A soma das 5 maiores causas para acidentes resultam em {soma_maiores:.2f}% dos acidentes de 2020.')

A soma das 5 maiores causas para acidentes resultam em 72.26% dos acidentes de 2020.


In [ ]:
causas_diferentes = len(dados.causa_acidente.unique())
causas_diferentes

58

Assim, 3.5 a cada 10 acidentes são causados por falta de atenção, evidenciando a importância de se reforçar políticas públicas que diminuam o uso do celular e outras distrações durante a condução do veículo.

Além disso, mesmo que tenhamos 58 tipos diferentes de causas, as 5 maiores concentram 72.26% dos acidentes ocorridos em 2020, mostrando que as políticas públicas devem, primariamente, conter acidentes causados por tais fatores.